sphinx/domains/python.py

```diff
@@ -788,9 +784,6 @@ def get_signature_prefix(self, sig: str) -> List[nodes.Node]:
         if 'classmethod' in self.options:
             prefix.append(nodes.Text('classmethod'))
             prefix.append(addnodes.desc_sig_space())
-        if 'property' in self.options:
```

Comment:
I noticed we should emit a warning for this usage to users during 5.x.

Reply:
Opened #10603 to track, we should add the warning in 5.0.3 or as early as possible otherwise.

A